ttnn/cpp/ttnn/operations/embedding_backward/device/embedding_backward_device_operation.cpp

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttnn/operations/embedding_backward/device/embedding_backward_device_operation.hpp"
+
+#include "tt_metal/common/constants.hpp"
+#include "ttnn/cpp/ttnn/run_operation.hpp"
+
+using namespace tt::constants;
+using namespace std;
+using namespace tt::tt_metal;
+
+namespace ttnn::operations::embedding_backward {
+
+void EmbeddingBackward::validate(const std::vector<Tensor> &input_tensors) const {
+    TT_FATAL(input_tensors.size() == 2, "Must have between 2 input tensors");
+
+    const auto &index_tensor = input_tensors.at(0);
+    const auto &grad_tensor = input_tensors.at(1);
+    const auto &index_tensor_shape = index_tensor.get_legacy_shape();
+    const auto &grad_tensor_shape = grad_tensor.get_legacy_shape();
+
+    TT_FATAL(
+        index_tensor.device()->arch() == tt::ARCH::WORMHOLE_B0,
+        "Embedding backwards is only implemented for Wormhole!");
+
+    TT_FATAL(index_tensor.get_layout() == Layout::ROW_MAJOR);
+    TT_FATAL(
+        index_tensor.get_dtype() == DataType::UINT32 or index_tensor.get_dtype() == DataType::BFLOAT16,
+        "Index tensor must be UINT32 or BFLOAT16");
+
+    TT_FATAL(
+        index_tensor_shape[1] == 1 && index_tensor_shape[2] == 1, "Only dim 0 && 3 for the index tensor can be non 1");
+
+    TT_FATAL(
+        index_tensor_shape[-1] % TILE_WIDTH == 0,
+        "Number of columns in the index tensor must be divisible by tile width");
```

Comment:
same idea. We need to include tile width abd shape

Reply:
Is there some convention i should follow?

Reply:
@TT-BrianLiu , https://github.com/tenstorrent/tt-metal/blob/main/ErrorMessageBestPractices.md